presto-accumulo/src/test/java/com/facebook/presto/accumulo/TestAccumuloDistributedQueries.java

```diff
@@ -0,0 +1,503 @@
+    }
+
+    @Override
+    public void testAddColumn()
+            throws Exception
+    {
+        try {
+            // TODO Adding columns via SQL are not supported until adding columns with comments are supported
+            super.testAddColumn();
+        }
+        catch (Exception e) {
+            assertEquals("Must have at least one non-row ID column", e.getMessage());
+        }
+    }
+
+    @Override
+    public void testCompatibleTypeChangeForView()
+            throws Exception
+    {
+        // Override because base class error: Must have at least one non-row ID column
+        assertUpdate("CREATE TABLE test_table_1 AS SELECT 'abcdefg' a, 1 b", 1);
+        assertUpdate("CREATE VIEW test_view_1 AS SELECT a FROM test_table_1");
+
+        assertQuery("SELECT a FROM test_view_1", "VALUES 'abcdefg'");
+
+        // replace table with a version that's implicitly coercible to the previous one
+        assertUpdate("DROP TABLE test_table_1");
+        assertUpdate("CREATE TABLE test_table_1 AS SELECT 'abc' a, 1 b", 1);
+
+        assertQuery("SELECT a FROM test_view_1", "VALUES 'abc'");
+
+        assertUpdate("DROP VIEW test_view_1");
+        assertUpdate("DROP TABLE test_table_1");
+    }
+
+    @Override
+    public void testCompatibleTypeChangeForView2()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_table_2 AS SELECT BIGINT '1' v, 2 w", 1);
+        assertUpdate("CREATE VIEW test_view_2 AS SELECT * FROM test_table_2");
+
+        assertQuery("SELECT v FROM test_view_2", "VALUES 1");
+
+        // replace table with a version that's implicitly coercible to the previous one
+        assertUpdate("DROP TABLE test_table_2");
+        assertUpdate("CREATE TABLE test_table_2 AS SELECT INTEGER '1' v, 2 w", 1);
+
+        assertQuery("SELECT v FROM test_view_2 WHERE v = 1", "VALUES 1");
+
+        assertUpdate("DROP VIEW test_view_2");
+        assertUpdate("DROP TABLE test_table_2");
+    }
+
+    @Override
+    public void testCreateTableAsSelect()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_create_table_as_if_not_exists (a bigint, b double)");
+        assertTrue(queryRunner.tableExists(getSession(), "test_create_table_as_if_not_exists"));
+        assertTableColumnNames("test_create_table_as_if_not_exists", "a", "b");
+
+        MaterializedResult materializedRows = computeActual("CREATE TABLE IF NOT EXISTS test_create_table_as_if_not_exists AS SELECT UUID() AS uuid, orderkey, discount FROM lineitem");
+        assertEquals(materializedRows.getRowCount(), 0);
+        assertTrue(queryRunner.tableExists(getSession(), "test_create_table_as_if_not_exists"));
+        assertTableColumnNames("test_create_table_as_if_not_exists", "a", "b");
+
+        assertUpdate("DROP TABLE test_create_table_as_if_not_exists");
+        assertFalse(queryRunner.tableExists(getSession(), "test_create_table_as_if_not_exists"));
+
+//        Function "UUID" not found
```

Comment:
Getting the below error when I make a new `AccumuloFunctionsPlugin` class.  I'll wait on getting these tests uncommented until the UUID function is migrated in the future -- I don't see value in adding another project for this.

```
[ERROR] You have more than one class that implements com.facebook.presto.spi.Plugin:
[ERROR] 
[ERROR] com.facebook.presto.accumulo.AccumuloFunctionsPlugin
[ERROR] com.facebook.presto.accumulo.AccumuloPlugin
[ERROR] 
[ERROR] You can only have one per plugin project.
```